src/cephadm/cephadm

```diff
@@ -5212,6 +5261,548 @@ def command_gather_facts():
 ##################################
 
 
+class CephadmCache:
+    task_types = ['disks', 'daemons', 'host', 'http_server']
+
+    def __init__(self):
+        self.started_epoch_secs = time.time()
+        self.tasks = {
+            "daemons": "inactive",
+            "disks": "inactive",
+            "host": "inactive",
+            "http_server": "inactive",
+        }
+        self.errors = []
+        self.disks = {}
+        self.daemons = {}
+        self.host = {}
```

Comment:
Minor improvement (I don't know the length of these), but if you store them (additionally) [gzipped](https://docs.python.org/3.8/library/gzip.html#gzip.compress), you can save some extra ms on each request. With dashboard we found this meant a noticeable improvement. And it's simple in terms of HTTP (just adding a 'Content-Encoding: gzip').